frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditDate/EditDate.tsx

```diff
@@ -216,6 +266,90 @@ export const EditDate = ({ field }: EditDateProps): JSX.Element => {
           {get(errors, 'dateValidation.customMinDate.message')}
         </FormErrorMessage>
       </FormControl>
+      <Stack>
+        <FormControl isReadOnly={isLoading}>
+          <Toggle
+            {...register('restrictParticularDays.addParticularDayRestriction')}
+            label="Customize days of the week"
+          />
+          <Text textStyle="body-2" color="secondary.400">
+            Checking a day will disable all the same days in the calendar
+          </Text>
+        </FormControl>
+        {getValues('restrictParticularDays.addParticularDayRestriction') ? (
+          <FormControl isRequired isReadOnly={isLoading}>
+            <Stack direction="row" spacing="0.75rem">
+              <Stack direction="column">
+                {PARTICULAR_DAYS_OPTIONS_LEFT.map((option) => {
```

Comment:
as discussed, rhf returning boolean is a bug when keys are nested due to the way rhf detects controlled checkboxes. 

either add and register a hidden checkbox with the same id, or use non-nested keys.

See https://github.com/react-hook-form/react-hook-form/issues/7834#issuecomment-1040735711.